client/src/endpoint/mod.rs

```diff
@@ -1,14 +1,15 @@
 //! This module contains the various end point definitions for stellar's horizon
 //! API server.
 use error::Result;
-use hyper::Uri;
 use serde::Deserialize;
+use http;
 
 mod account;
 pub use self::account::AccountDetails;
 
 pub(crate) trait EndPoint<'de> {
     type Response: Deserialize<'de>;
+    type RequestBody;
```

Comment:
Want to make sure I'm understanding this correctly.  An Endpoint must have a response that implements deserialize.  It must also have a request body that can be anything.

Reply:
I like this a lot.

Reply:
Yup, exactly.

We may want to put a limit on the request body but there is no limit on it in the http crate at the moment. I'm not sure how we'd capture a `Serialize` trait on it while also allowing optional. Maybe there's a way :)